### PR TITLE
[NPUW]Fixed V tensor transpose for shared KV-cache in Gemma4.

### PR DESCRIPTION
**Pull request by intelgaoxiong**
### Details:
When the same past_value->concat is consumed by two matmul branches, GraphRewrite fires the MHA matcher once per matmul.
The shared mutations (param shape swap, transpose order, concat axis) must be applied exactly once.
Otherwise, we would see error due to shape mismatch.

```
While validating node 'opset1::Concat __module.model.language_model.layers.40.self_attn/aten::to/Convert (opset1::Parameter past_key_values.22.value[0]:f32[1,2,383,256], opset1::Transpose __module.model.language_model.layers.22.self_attn/aten::transpose/Transpose_3[0]:f32[1,2,256,1]) -> (f32[1,2,384,256])' with friendly_name '__module.model.language_model.layers.40.self_attn/aten::to/Convert':
Shape inference input shapes {[1,2,383,256],[1,2,256,1]}
Argument shapes are inconsistent; they must have the same rank, and must have equal dimension everywhere except on the concatenation axis (axis 3).
```

### Tickets:
 - *[EISW-211488](https://jira.devtools.intel.com/browse/EISW-211488)*

### AI Assistance:
 - *AI assistance used: no / yes*
 - *If yes, summarize how AI was used and what human validation was performed (build/tests/manual checks).*

### CODE REVIEW

File: src/plugins/intel_npu/tests/unit/npuw/transpose_vt.cpp
Comment (intelgaoxiong) [inline src/plugins/intel_npu/tests/unit/npuw/transpose_vt.cpp:1]: Just added a new test case.
Others are changed by clang-format.

**Review by dmatveev — COMMENTED** [review]
This change impacts the existing transformations so must be thoroughly tested as well. @esmirno you're the owner here, please work with Xiong to ensure the right test coverage.

**Comment by intelgaoxiong** [comment]
> This change impacts the existing transformations so must be thoroughly tested as well. @esmirno you're the owner here, please work with Xiong to ensure the right test coverage.

Running a full validation in https://cje-ir-prod01.devtools.intel.com/sai-npu-experience/job/Staging/job/xiong/job/Validate/21/

File: src/plugins/intel_npu/tests/unit/npuw/transpose_vt.cpp [hunk 1]
```diff
@@ -363,16 +400,81 @@ const std::vector<NetworkKind> networkKind = {
...
+// Model topology:
+//
+//   param[1,4,8,64]   param2[1,1,256]
+//        |                  |
+//        |           reshape[1,1,4,64]
+//        |                  |
+//        |           transpose{0,2,1,3}[1,4,1,64]
+//        +-------concat(axis=2)[1,4,9,64]--------+
+//                      |                          |
+//              softmax1->matmul1          softmax2->matmul2
```
Comment (esmirno) [inline src/plugins/intel_npu/tests/unit/npuw/transpose_vt.cpp:431]: @intelgaoxiong it is actually interesting pattern - how SDPA detector will work in that case - for every SDPA pattern matlum->softmax->matmul  that shared softmax node - will it return same concat? If so then in dynamic-dequantize phase we need to detect that situation, to avoid multiple DQ

**Comment by esmirno** [comment]
regarding validation i dont think we will hit introduced if-case even right? but we need to have this sort of models to test different combinations of shared-concats for example eliminate empty-inputs pass also might fail

File: src/plugins/intel_npu/tests/unit/npuw/transpose_vt.cpp (hunk 1 above)
Comment (intelgaoxiong) [reply src/plugins/intel_npu/tests/unit/npuw/transpose_vt.cpp:431]: @esmirno The particular pattern is introduced by Gemma4 model feature - sharing KV cache with some layers.
SDPA detector can't work for the pattern for sure.
For Day 0 model support, we focus on 1K case.
Other optimization has not be applied.

**Comment by intelgaoxiong** [comment]
> regarding validation i dont think we will hit introduced if-case even right? but we need to have this sort of models to test different combinations of shared-concats for example eliminate empty-inputs pass also might fail

@esmirno 
1. Actually, `TEST(TransposeVTSharedConcatTest, smoke_SharedConcat_MHA_MultipleUsers)` hits the PR changes. Without the fix in PR, shape mismatch error would appear.
2. `RemoveEmptyKVTensors` works fine for the shared-kv pattern: it redirects second concat input to every node which reads from concat, see [here](https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_npu/src/plugin/npuw/embedding/remove_empty_kv_inputs.cpp#L90). Anyway, I have added one more test case for this.

**Review by esmirno — APPROVED** [review]
LGTM

**Comment by intelgaoxiong** [comment]
@dmatveev @esmirno May I know if this PR is OK to be merged?
Thanks!

**Review by dmatveev — APPROVED** [review]
